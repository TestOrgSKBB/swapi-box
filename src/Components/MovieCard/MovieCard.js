import React from 'react';
import './MovieCard.scss';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const MovieCard = (props) => {
  const { title, release_date, id, episode_id, selectMovie } = props;
  return (
    <section className="section__movie--card" id={episode_id} >
      <h2 className="movie__card--element movie__element--title">{title}</h2>
      <h3 className="movie__card--element">Episode {id}</h3>
      <h3 className="movie__card--element">{release_date}</h3>
      <Link className="movie__button--link" to={`movies/${id}/characters`}>
        <button 
          className="button_view-characters" 
          value={props.episode} 
          onClick={selectMovie}
        >
          View Characters
        </button>
      </Link>
    </section>
  );
};

MovieCard.propTypes = {
  characters: PropTypes.array,
  episode_id: PropTypes.number,
  id: PropTypes.number,
  opening_crawl: PropTypes.string,
  release_date: PropTypes.string,
  selectMovie: PropTypes.func,
  title: PropTypes.string
};


export default MovieCard