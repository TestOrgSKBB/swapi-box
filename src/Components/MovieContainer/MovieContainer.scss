.section__movie--container {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
}

.movies-title {
    color: #919192;
    font-size: 3em;
    position: absolute;
    top: 0;
}