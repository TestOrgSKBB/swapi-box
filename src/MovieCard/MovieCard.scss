.section__movie--card {
  color: white;
  background: rgba(145, 145, 145, 0.7);
  color: white;
  width: 20%;
  margin: 2%;
  height: 52vh;
  text-align: center;
  position: relative;
  border-radius: 2em;
}

.button_view-characters {
  background: rgba(32, 37, 40, .5);
  color: white;
  height: 6vh;
  // border-radius: 2em;
  font-size: 1.25em;
  width: 100%;
  position: absolute;
  bottom: 0;
  left: 0;
  height: 7vh;
  border: 1px solid rgba(32, 37, 40, .8)
}

.button_view-characters:hover {
  background:rgba(32, 37, 40, .8);
  font-size: 1.5em;
  cursor: pointer;
  color: rgba(143, 171, 172, 0.7);
  text-shadow: .5px .5px white;
}

.movie__card--element {
  margin-top: 15%;
  background: rgba(51, 51, 51, .5);
  font-size: 1.5em;
  box-shadow: 0px 2px rgba(32, 37, 40, .8);
}

.movie__element--title {
  color: rgba(143, 171, 172, 0.7);
  text-shadow: 1px 2px rgba(32, 37, 40, 0.9);
  font-size: 1.75em;
}