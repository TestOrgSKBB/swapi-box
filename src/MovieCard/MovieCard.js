import React from 'react';
import './MovieCard.scss';
import { Route, NavLink } from 'react-router-dom'
// import MovieCard from '../MovieCard/MovieCard';

const MovieCard = (props) => {
  const { title, release_date, episode_id } = props;
  return (
    <section>
      <h2>{title}</h2>
      <h3>{episode_id}</h3>
      <h3>{release_date}</h3>
      <NavLink className="movie__button--link" to='/movies/3'>
      <button value={props.episode} onClick={event => props.selectMovie(event.target.value)}>>View Characters</button>
      </NavLink>
    </section>
  );
}


export default MovieCard