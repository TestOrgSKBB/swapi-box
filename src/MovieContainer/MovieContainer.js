import React from 'react';
import MovieCard from '..MovieCard/MovieCard';

const MovieContainer = ({movies}) => {
  const movieInfo = movies.map(movie => {
    return <MovieCard 
      id={movie.episode_id}
      title= {movie.title}
      episode={movie.episode_id}
      year={movie.release_date}
      />
  })
}

export default MovieContainer;