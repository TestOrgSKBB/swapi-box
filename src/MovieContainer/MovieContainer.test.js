import React from 'react';
import { shallow } from 'enzyme'
import MovieContainer from './MovieContainer';
import MovieCard from '../MovieCard/MovieCard';

describe('MovieContainer', () => {
  let wrapper;
  const movies = [{
    title: 'Great Movie',
    episode_id: 4,
    release_date: '1999/04/20', 
  },
  {
    title: 'First Star Wars',
    episode_id: 1,
    release_date: '2050/01/01',
  }];

  beforeEach(() => {
    wrapper = shallow(<MovieContainer movies={movies} />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
});