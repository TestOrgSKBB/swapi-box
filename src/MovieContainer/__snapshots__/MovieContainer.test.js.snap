// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should match the snapshot 1`] = `
<Fragment>
  <h1>
    Movies
  </h1>
  <MovieCard
    episode_id={4}
    key="0"
    release_date="1999/04/20"
    title="Great Movie"
  />
  <MovieCard
    episode_id={1}
    key="1"
    release_date="2050/01/01"
    title="First Star Wars"
  />
</Fragment>
`;
